app/react/V2/Routes/Settings/Thesauri/ThesaurusForm.tsx

```diff
@@ -1,212 +1,70 @@
-        const newValues = singleThesaurus.values?.filter(
-          currentGroupItem =>
-            !selectedThesaurusValue.find(selected => selected.original._id === currentGroupItem._id)
-        );
-        singleThesaurus.values = newValues;
-      }
-      return singleThesaurus;
-    });
-
-    setThesaurusValues(childrenDeleted);
-  };
-
-  const sortValues = () => {
-    const valuesCopy = [...thesaurusValues];
-    valuesCopy.sort((first, second) => (first.label > second.label ? 1 : -1));
-    setThesaurusValues(valuesCopy);
-  };
+    formState: { errors },
+  } = form;
 
-  const addItemSubmit = (items: FormThesauriValue[]) => {
-    const addingNewItems = !items[0]._id;
-    const thesaurusValuesCopy = [...thesaurusValues];
-    if (addingNewItems) {
-      items.forEach(item => {
-        const itemWithId = { ...item, _id: `temp_${uniqueID()}` };
-        if (!itemWithId.groupId) {
-          thesaurusValuesCopy.push(itemWithId);
-          return;
-        }
-
-        const group = thesaurusValuesCopy.find(groupItem => groupItem._id === itemWithId.groupId);
-        if (group) {
-          group.values = group.values || [];
-          group.values.push(itemWithId);
-        }
-      });
+  const navigate = useNavigate();
+  const [thesauri, setThesauri] = useAtom(thesauriAtom);
+  const setNotifications = useSetAtom(notificationAtom);
 
-      setThesaurusValues(thesaurusValuesCopy);
-      return;
+  const saveThesaurus = async (data: ClientThesaurus) => {
+    const thesaurusToUpdate = { ...data, values: sanitizeThesaurusValues(thesaurusValues) };
+    const savedThesaurus = await ThesauriAPI.save(thesaurusToUpdate);
+    if (thesauri.find(item => item._id === savedThesaurus._id)) {
+      remove(thesauri, item => item._id === savedThesaurus._id);
     }
-
-    items.forEach(item => {
-      if (!item.groupId) {
-        const index = thesaurusValuesCopy.findIndex(tv => tv._id === item._id);
-        thesaurusValuesCopy[index] = item;
-        return;
-      }
-
-      const group = thesaurusValuesCopy.find(groupItem => groupItem._id === item.groupId);
-      if (group) {
-        const index = group.values?.findIndex(tv => tv._id === item._id);
-        if (index !== undefined && index !== -1) {
-          group.values = group.values || [];
-          group.values[index] = item;
-        }
-      }
+    thesauri.push(savedThesaurus);
+    setThesauri([...thesauri]);
+    setValue('_id', savedThesaurus._id);
+    setNotifications({
+      type: 'success',
+      text: thesaurus ? (
+        <Translate>Thesauri updated.</Translate>
+      ) : (
+        <Translate>Thesauri added.</Translate>
+      ),
     });
-    setThesaurusValues(thesaurusValuesCopy);
-  };
-
-  const addGroupSubmit = (group: FormThesauriValue) => {
-    const thesaurusValuesCopy = thesaurusValues.filter(item => item._id !== group._id);
-    setThesaurusValues([...thesaurusValuesCopy, group]);
+    navigate(`../edit/${savedThesaurus._id}`);
```

Comment:
Is this to reload the page? We have something similar here 
https://github.com/huridocs/uwazi/blob/cde642be4baffd54d9876c182d229ff08d58a900/app/react/V2/Routes/Settings/Pages/PageEditor.tsx#L86

The example also prevents a problems where if the user navigates back they would end up in the new thesaurus screen in some cases.